released_plugins/v3d_plugins/MIND/soma_segmentation_plugin.cpp

```diff
@@ -200,28 +200,63 @@ void reconstruction_func(V3DPluginCallback2 &callback, QWidget *parent,
     c = PARA.channel;
   }
 
-  // main neuron reconstruction code
-
   //// THIS IS WHERE THE DEVELOPERS SHOULD ADD THEIR OWN NEURON TRACING CODE
 
-  // Output
-  NeuronTree nt;
-  QString swc_name = PARA.inimg_file + "_soma_segmentation.swc";
-  nt.name = "soma_segmentation";
-  writeSWC_file(swc_name.toStdString().c_str(), nt);
+  // get current window, image, and landmarks
+  v3dhandle curwin = callback.currentImageWindow();
+  Image4DSimple *p4DImage = callback.getImage(curwin);
+  LandmarkList landmarkList = callback.getLandmark(curwin);
 
-  if (!bmenu) {
-    if (data1d) {
-      delete[] data1d;
-      data1d = 0;
-    }
+  // reset ROI
+  ROIList roiList = callback.getROI(curwin);
+  for (int j = 0; j < 3; j++) {
+    roiList[j].clear();
+  }
+
+  // get 1st landmark
+  LocationSimple lm = landmarkList[0];
+  float x, y, z;
+  float radius;
+  x = lm.x;
+  y = lm.y;
+  z = lm.z;
+  radius = lm.radius;
+
+  v3d_msg(QString("Landmark: x=%1, y=%2, z=%3, radius=%4")
+              .arg(x)
+              .arg(y)
+              .arg(z)
+              .arg(radius),
+          bmenu);
+
+  // set ROI
+  float x_min = x - radius * 2.0f;
+  float x_max = x + radius * 2.0f;
+  float y_min = y - radius * 2.0f;
+  float y_max = y + radius * 2.0f;
+  float z_min = z - radius * 2.0f;
+  float z_max = z + radius * 2.0f;
+  roiList[0] << QPoint(x_min, y_min);
+  roiList[0] << QPoint(x_max, y_min);
+  roiList[0] << QPoint(x_max, y_max);
+  roiList[0] << QPoint(x_min, y_max);
+  roiList[1] << QPoint(z_min, y_min);
+  roiList[1] << QPoint(z_max, y_min);
+  roiList[1] << QPoint(z_max, y_max);
+  roiList[1] << QPoint(z_min, y_max);
+  roiList[2] << QPoint(x_min, z_min);
+  roiList[2] << QPoint(x_max, z_min);
+  roiList[2] << QPoint(x_max, z_max);
+  roiList[2] << QPoint(x_min, z_max);
```

Comment:
Can we add comments here explaining what the roiList[n] << QPoint() is doing? It isn't quite clear to me

Reply:
I have added some, please let me know if it is clear